Src/VimCore/InsertUtil.fs

```diff
@@ -208,70 +208,46 @@ type internal InsertUtil
                     let span = column.Column.Span
                     textEdit.Replace(span.Span, text) |> ignore
 
-            if textEdit.HasEffectiveChanges then 
+            if textEdit.HasEffectiveChanges then
                 textEdit.Apply() |> ignore
             else
                 textEdit.Cancel()
 
-        // Actually apply the edit to the lines in the block selection
-        let doApply () = 
-            use textEdit = _textBuffer.CreateEdit()

```

Comment:
This line gave me a bit of pause. I'm trying to remember why I implemented this as a direct `ITextEdit` operation vs. a series of `ITextBuffer` edit operations. 

The only thing I can think of off the top of my head is perf / simplicity. The nice part of `ITextEdit` is that we don't have to worry about other extensions doing edit in between every single line change. That probably isn't a big deal though here. Also doing it as a single edit makes perf "better" but probably not discernably so.

Mostly commenting here so others are aware of this in case it comse up again. I can't see anything wrong with this but it tickles my brain a bit.

Reply:
At the moment, there is no way to request that insertion command execution occur in the context of a text edit, and as a result, there is no easy way to batch insertion commands other than to make sure that they are inside of an enclosing undo transaction. But, they already are.

In this particular case, the repeat loop for each  subsequent line in the block insert correctly uses no state information from the previous snapshot and so benefit of performing all the edits against a single snapshot is unnecessary.

It could be argued that a single text edit is more efficient, but given that insert commands currently operate on the latest snapshot, it would require a major reorganization to try to achieve a very modest performance gain in a fairly uncommon insertion use case.

The most important thing is that undo undoes the whole block insertion, and it does because of the enclosing undo transaction.